test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageIdValidatorTest.cs

```diff
@@ -18,6 +18,23 @@ public void ValidatePackageIdInvalidIdThrows()
             ExceptionAssert.ThrowsArgumentException(() => PackageIdValidator.ValidatePackageId(packageId), "The package ID '  Invalid  . Woo   .' contains invalid characters. Examples of valid package IDs include 'MyPackage' and 'MyPackage.Sample'.");
         }
 
+        [Fact]
+        public void ValidatePackageId_OnGB18030Characters_Succeeds()
+        {
+            // Arrange
+            string packageId = "G龦鿃C啊阿鼾齄D2丂丄狚狛D3狜狝﨨﨩D4ⅫㄨㄩD1ˊˋ〇D5U1U2U3㐀㐁䶴䶵ABCDEFKS";
+
+            // Act & Assert
+            try
+            {
+                PackageIdValidator.ValidatePackageId(packageId);
+            }
+            catch (Exception ex)
+            {
+                Assert.Fail(ex.Message);
+            }
```

Comment:
We shouldn't use try/catch blocks in tests. It seems that you may be trying to validate that an exception will be thrown when calling `PackageIdValidator.ValidatePackageId(packageId);` with the specified input. If that's the case, you'll want to use `Assert.Throws<TypeOfExceptionHere>(() => PackageIdValidator.ValidatePackageId(packageId));` That will make sure that the test verifies that this code path, given the input provided, will always throw an exception.

The try/catch block we have in the test right now will only execute the assertion if the method under tests actually throws an exception. If it doesn't, the test will do nothing and not assert anything which could obscure a bug.

Reply:
It's trying to assert an exception is not thrown rather

Reply:
I think we should do what JeanPierre is suggesting, if I'm not wrong if by any chance the code changes and `PackageIdValidator.ValidatePackageId(packageId);` doesn't throw the test will succeed and won't catch the exception assert.